lib/lifecycle.ml

```diff
@@ -1,3 +1,5 @@
 (* No shutdown events on Solo5. *)
 let await_shutdown_request ?(can_poweroff:_) ?(can_reboot:_) () =
+  let _poweroff = can_poweroff in
+  let _rebooot = can_reboot in
```

Comment:
It would be better to write:

```ocaml
let await_shutdown_request ?can_poweroff:_ ?can_reboot:_ () =
```

e.g. the current form ignore the type, but what you want is ignoring the values (yes, that syntax is confusing).